src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -163,13 +181,23 @@ internal partial class DiagnosticIncrementalAnalyzer
                     }
 
                     // PERF: Check whether we want to analyze this project or not.
-                    if (!FullAnalysisEnabled(project, forceAnalyzerRun))
+                    if (!FullAnalysisEnabled(project, forceAnalyzerRun, out var compilerFullAnalysisEnabled, out var analyzersFullAnalysisEnabled))
```

Comment:
consider three out params.  i'm not really sure what this means.

Reply:
The API returns true if full solution analysis is enabled for either the compiler diagnostics OR analyzer diagnostics. Even if one of these is not full solution enabled, we still need to analyze the entire solution to compute solution wide diagnostics for the other. Having a third out parameter here instead of the API returning the OR of the out flags would lead to a verbose pattern at callsites that only care about whether or not full solution is going to be analyzed. We can chat more offline if you still disagree.